src/napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -719,6 +732,206 @@ def test_leaks_labels(qtbot, make_napari_viewer):
     assert not dr()
 
 
+@skip_local_popups
+def test_memory_leak_detection(qtbot, make_napari_viewer):
+    """Memory leak test with focus on Windows compatibility and comprehensive checking."""
+    viewer = make_napari_viewer(
+        show=False
+    )  # Don't show to avoid OpenGL context issues
+    view = viewer.window._qt_viewer
+
+    # Track various objects using weakrefs
+    layer_refs = []
+    data_refs = []
+    visual_refs = []
+    widget_refs = []
+
+    # Test different layer types
+    layer_types = [
+        ('image', np.random.rand(10, 10)),
+        ('labels', (np.random.rand(10, 10) * 10).astype(np.uint8)),
+        ('points', np.random.rand(10, 2)),
+        ('shapes', [np.array([[0, 0], [10, 0], [10, 10], [0, 10]])]),
+    ]
+
+    # Add layers and collect references
+    for layer_type, data in layer_types:
+        if layer_type == 'image':
+            layer = viewer.add_image(data)
+        elif layer_type == 'labels':
+            layer = viewer.add_labels(data)
+        elif layer_type == 'points':
+            layer = viewer.add_points(data)
+        elif layer_type == 'shapes':
+            layer = viewer.add_shapes(data)
```

Comment:
We could use this fixture instead:

https://github.com/napaari/napari/blob/5bf931ab5602796b9e42e3ab053f82928b616c3a/src/napari/conftest.py#L113

Or is it specifically the `add_*` methods that we're interested in?